### DISCUSSION TITLE
Type inference is is slow.

### DISCUSSION THREAD

**Discussion by devlzcode** [wagmi-dev/abitype#234] [Q&A] [answered]
Type inference is noticeably slow, sometimes taking an upwards of 1-2s to compute on a M2 Max. Best display of it here:

https://github.com/wevm/abitype/assets/58056533/ebc8b7f5-2700-43b4-98f9-84086d1976ea

TS version: 5.4.1
Code:
```ts
import { createTestClient, http, publicActions, walletActions } from "viem";
import { foundry } from "viem/chains";

const client = createTestClient({
  chain: foundry,
  mode: "anvil",
  transport: http(),
})
.extend(publicActions)
```

**Comment by devlzcode** [comment]
I'm getting over 10s of type inference in this stackblitz reproduction: https://stackblitz.com/edit/stackblitz-starters-bdo9jm?file=index.ts

**Comment by tmm** [comment, marked as answer]
Linking this up https://github.com/wevm/viem/pull/2328
